### ISSUE TITLE
Problem with haskell version of bnfc

### ISSUE THREAD

**Issue by sirdude** [BNFC/bnfc#149]
The following grammar works with bnfc -c or --cpp but does not build with haskell

Prog.           Program ::= Defin ;
Pe.             Program ::= "" ;
Def.            Defin   ::= "woo" ;

It will generate files but when trying to make I get the following:
% make
happy -gca ParTest.y
alex -g LexTest.x
LexTest.x:25:3: parse error

**Comment by gdetrez** [comment]
BNFC chokes on empty terminals. If you want an empty production do this instead:
```
Prog. Program ::= Defin ;
Pe. Program ::= ;
Def. Defin ::= "woo" ;
```

**Comment by gdetrez** [comment]
Fixed in a3a41499a6421ad3f9eb6f5e208760a15632e084.
